aten/src/ATen/native/TensorIterator.cpp

```diff
@@ -682,6 +682,8 @@ TensorIterator TensorIterator::reduce_op(Tensor& out, const Tensor& a) {
   iter.promote_gpu_output_dtypes_ = true;
   iter.resize_outputs_ = false;
   iter.is_reduction_ = true;
+  // TODO: This is only really necessary for arg{min,max}
+  iter.compute_common_dtype_only_for_inputs();
```

Comment:
There are some reductions where the input has to be cast to the output `dtype` e.g. `mean`. So, there needs to be at least two versions here. What would be the best way to do that: add a parameter/function just about `dtype`s, or an option to return the iterator without calling `build()` first?